tamr_client/_types/backup.py

```diff
@@ -0,0 +1,24 @@
+from dataclasses import dataclass
+
+from tamr_client._types.url import URL
+
+
+@dataclass(frozen=True)
+class Backup:
+    """A Tamr backup
+
+    See https://docs.tamr.com/new/docs/configuration-backup-and-restore
+
+    Args:
+        url
+        resource_id
+        path
+        state
+        error_message
+    """
+
+    url: URL
+    resource_id: str
```

Comment:
I don't think we want to model `resource_id` explicitly, but rather rely on the `url` field instead.

```suggestion
```

Note: should be removed from docstring as well.

Reply:
I think this is reasonable to avoid redundancy (and possible conflicts in user-constructed objects).  This does mean it is moderately harder to get an updated version of a `Backup`, since you can no longer use `backup = tc.backup.by_resource_id(s, instance, backup.resource_id)`.  It isn't much harder to do `str(backup.url).split("/")[-1]`, but maybe this change justifies adding a `poll` function for this purpose.

Reply:
Rather than the error-prone `resource_id` field, we could either provider a `resource_id(backup)`. But semantically, you should be able to just pass in the `backup` to a function to get what you want.

I.e. why are you even getting the resource ID? To poll the backup? Then just call `poll(backup)`.

So yes, I think this justifies a `poll` function 😄